modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -754,6 +754,159 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
+			errorReceiptProof, proofHeight = suite.chainB.QueryProof(upgradeErrorReceiptKey)
+
+			var ok bool
+			errorReceipt, ok = suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+			suite.Require().True(ok)
+
+			tc.malleate()
+
+			err := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeCancel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, errorReceipt, errorReceiptProof, proofHeight)
+
+			expPass := tc.expError == nil
+			if expPass {
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().ErrorIs(err, tc.expError)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestWriteUpgradeCancelChannel() {
+	suite.SetupTest()
+
+	path := ibctesting.NewPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path)
+
+	path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+	path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+
+	suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
+
+	// cause the upgrade to fail on chain b so an error receipt is written.
+	suite.chainB.GetSimApp().IBCMockModule.IBCApp.OnChanUpgradeTry = func(
+		ctx sdk.Context, portID, channelID string, order types.Order, connectionHops []string, counterpartyVersion string,
+	) (string, error) {
+		return "", fmt.Errorf("mock app callback failed")
+	}
+
+	suite.Require().NoError(path.EndpointB.ChanUpgradeTry())
+
+	suite.Require().NoError(path.EndpointA.UpdateClient())
+
+	errorReceipt, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+	suite.Require().True(ok)
+
+	suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeCancelChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, errorReceipt.Sequence)
+
+	channel := path.EndpointA.GetChannel()
+
+	// Verify that channel has been restored to previous state
+	suite.Require().Equal(types.OPEN, channel.State)
+	suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus)
+	suite.Require().Equal(mock.Version, channel.Version)
+	suite.Require().Equal(errorReceipt.Sequence, channel.UpgradeSequence)
+
+	// Assert that state stored for upgrade has been deleted
+	upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+	suite.Require().Equal(types.Upgrade{}, upgrade)
+	suite.Require().False(found)
+	lastPacketSequence, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyLastPacketSequence(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+	suite.Require().Equal(uint64(0), lastPacketSequence)
+	suite.Require().False(found)
```

Comment:
whitespacing again :D 

```suggestion
	upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
	suite.Require().Equal(types.Upgrade{}, upgrade)
	suite.Require().False(found)
	
	lastPacketSequence, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyLastPacketSequence(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
	suite.Require().Equal(uint64(0), lastPacketSequence)
	suite.Require().False(found)
```